ale/base/data_naif.py

```diff
@@ -581,15 +581,29 @@ def sensor_position(self):
                     rotated_state = spice.mxvg(matrix, state)
                     states.append(rotated_state)
             else:
-                kwargs = {"target": target,
-                          "observer": observer,
-                          "frame": self.reference_frame,
-                          "abcorr": self.light_time_correction,
-                          "mission": self.spiceql_mission,
-                          "searchKernels": self.search_kernels}
-                states = spiceql_access.get_ephem_data(ephem, "getTargetStates", web=self.use_web, function_args=kwargs)
-                states = np.array(states)[:,0:6]
-
+                if isinstance(self, LineScanner) and self.use_web:
+                    logger.debug("Sensor is a Line Scanner, using alt API")
+                    start_ets, stop_ets, exposure_durations = self.exposure_rates
+                    kwargs = {"startEts": start_ets,
+                              "stopEts": stop_ets,
+                              "exposureDuration": exposure_durations,
+                              "target": target,
+                              "observer": observer,
+                              "frame": "J2000",
+                              "abcorr": self.light_time_correction,
+                              "mission": self.spiceql_mission,
+                              "searchKernels": self.search_kernels}
+                    states = self.spiceql_call("getTargetStates", web=self.use_web, function_args=kwargs)
```

Comment:
We should be able to use this for framers, line scanners, and push broom cameras. I guess I don't know why we are only going to use it for line scanners